app/javascript/hyacinth_ui_v1/components/projects/permissions/PermissionsEditor.js

```diff
@@ -0,0 +1,255 @@
+    getUsersQuery,
+  );
+  const { loading: permissionsLoading, error: permissionsError } = useQuery(
+    getProjectPermissionsQuery, {
+      variables: { stringKey: projectStringKey },
+      onCompleted: (data) => {
+        setProjectPermissions(data.projectPermissionsForProject);
+      },
+    },
+  );
+
+  const [updateProjectPermissions, { error: updateProjectPermissionsError }] = useMutation(updateProjectPermissionsMutation);
+
+  if (actionsLoading || permissionsLoading || usersLoading) return (<></>);
+  if (actionsError || permissionsError || usersError) {
+    return (<GraphQLErrors errors={actionsError || permissionsError || usersError || updateProjectPermissionsError} />);
+  }
+
+  const actionToDisplayLabel = (action) => {
+    return action.split('_').map(part => part.charAt(0).toUpperCase() + part.slice(1)).join(' ');
+  };
+
+  const updatePermissionsData = (userId, action, enabled) => {
+    setPermissionsChanged(true);
+
+    const { projectPermissionActions: { actions, manageAction } } = actionsData;
+
+    setProjectPermissions(
+      produce(projectPermissions, (draft) => {
+        const { permissions } = draft.find(
+          projectPermission => projectPermission.user.id === userId,
+        );
+        if (enabled) {
+          if (action === manageAction) {
+            // disable any current actions
+            permissions.splice(0, permissions.length);
+            // enable all actions
+            permissions.push(...actions);
+          } else {
+            // enable single action
+            permissions.push(action);
+          }
+        } else {
+          // disable single action
+          permissions.splice(permissions.indexOf(action), 1);
+        }
+      }),
+    );
+  };
+
+  const removeUser = (userId) => {
+    setPermissionsChanged(true);
+    setProjectPermissions(
+      produce(projectPermissions, (draft) => {
+        arrRemove(draft, (projectPermission) => {
+          return projectPermission.user.id === userId;
+        });
+      }),
+    );
+    setRemovedUserIds(
+      produce(removedUserIds, (draft) => {
+        draft.add(userId);
+      }),
+    );
+  };
+
+  const renderProjectPermission = (projectPermission) => {
+    const { projectPermissionActions: { actions, readObjectsAction } } = actionsData;
+    return actions.map(action => (
+      <td key={action}>
+        { /* eslint-disable jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for */ }
+        { /* Note 1: ESLint is getting confused by label + react-bootstrap element. */ }
+        { /* Note 2: Using a label for a wider clickable area. */ }
+        <label id={`label-${projectPermission.user.id}-${action}`} htmlFor={`checkbox-${projectPermission.user.id}-${action}`} className="w-100">
+          <Form.Check
+            id={`checkbox-${projectPermission.user.id}-${action}`}
+            label="&nbsp;"
+            disabled={readOnly || action === readObjectsAction}
```

Comment:
Can ticket separately, but we should probably also disable the create permission widget unless `project.is_primary`.

Reply:
Good call.  Will do in an upcoming commit in this PR!